Tests/MongoSwiftTests/AsyncAwaitTestUtils.swift

```diff
@@ -51,6 +51,40 @@ func assertIsEventuallyTrue(
     XCTFail("Expected condition \"\(description)\" to be true within \(timeout) seconds, but was not")
 }
 
+/// Asserts that the provided block returns true within the specified timeout. Nimble's `toEventually` can only be used
+/// rom the main testing thread which is too restrictive for our purposes testing the async/await APIs.
+@available(macOS 10.15.0, *)
+func assertIsEventuallyTrue(
+    description: String,
+    timeout: TimeInterval = 5,
+    sleepInterval: TimeInterval = 0.1,
+    _ block: @escaping () async throws -> Bool
+) async throws {
+    // Task that does the work as long as its not cancelled
+    let workTask = Task { () -> Bool in
+        while !Task.isCancelled {
+            guard try await block() else {
+                try? await Task.sleep(seconds: sleepInterval)
```

Comment:
why do we need to `try?` here instead of just `try`?

Reply:
oooo ok. can we add a comment saying that this is to ignore `CancellationError`s then?